src/Take.Blip.Builder/InputExpirationHandler.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Lime.Messaging.Contents;
+using Lime.Protocol;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Builder.Content;
+using Take.Blip.Builder.Models;
+using Take.Blip.Client.Content;
+using Take.Blip.Client.Extensions.Scheduler;
+
+namespace Take.Blip.Builder
+{
+    public class InputExpirationHandler : IInputExpirationHandler
+    {
+        private readonly Document _emptyContent = new PlainText() { Text = string.Empty };
+        private readonly ISchedulerExtension _schedulerExtension;
+        private string _stateIdInputExipiration = null;
```

Comment:
Creating state variables in a singleton instance can be a problem since all the communication it's async and the single lock it has is with the contact(can have collisions with other contacts).